components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Engine/LegacyJobOrchestratorTests.cs

```diff
@@ -0,0 +1,56 @@
+// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Tests.Engine {
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Agent.Framework;
+    using Agent.Framework.Models;
+    using Diagnostics;
+    using Models;
+    using Module;
+    using Moq;
+    using Publisher.Engine;
+    using Serializers.NewtonSoft;
+    using Xunit;
+    using static Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.Agent.PublisherJobsConfiguration;
+
+    /// <summary>
+    /// Tests the LegacyJobOrchestrator class
+    /// </summary>
+    public class LegacyJobOrchestratorTests {
+
+        [Fact]
+        public async Task GetAvailableJobAsyncMulithreading() {
+            var legacyCliModelProviderMock = new Mock<ILegacyCliModelProvider>();
+            var agentConfigProviderMock = new Mock<IAgentConfigProvider>();
+            var identityMock = new Mock<IIdentity>();
+            var newtonSoftJsonSerializer = new NewtonSoftJsonSerializer();
+            var jobSerializer = new PublisherJobSerializer(newtonSoftJsonSerializer);
+            var publishedNodesJobConverter = new PublishedNodesJobConverter(TraceLogger.Create(), newtonSoftJsonSerializer);
+
+            var legacyCliModel = new LegacyCliModel { PublishedNodesFile = ".\\Engine\\publishednodes.json"};
```

Comment:
@oliverchristen, please change the code so that tests also work for Linux and Mac. Currently, this is causing the following error:

```
[2020-09-16 09:59:51Z ERR Root] Error while loading job from file. Retry expired, giving up. 
System.IO.FileNotFoundException: Could not find file '/home/vsts/work/1/s/components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/bin/Release/netcoreapp3.1/.\Engine\publishednodes.json'.
File name: '/home/vsts/work/1/s/components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/bin/Release/netcoreapp3.1/.\Engine\publishednodes.json'
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at System.IO.File.OpenRead(String path)
   at Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Engine.LegacyJobOrchestrator.GetChecksum(String file) in /home/vsts/work/1/s/components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs:line 128
   at Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Engine.LegacyJobOrchestrator.RefreshJobFromFile() in /home/vsts/work/1/s/components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs:line 139
[2020-09-16 09:59:51Z INF Root] Read 4 items from published nodes file in 00:00:00.0081637 
[2020-09-16 09:59:51Z INF Root] Converted items to jobs in 00:00:00.0005461 
[xUnit.net 00:00:01.20]     Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Tests.Engine.LegacyJobOrchestratorTests.GetAvailableJobAsyncMulithreading [FAIL]
[2020-09-16 09:59:52Z INF Root] Read 1 items from published nodes file in 00:00:00.0546910 
  X Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Tests.Engine.LegacyJobOrchestratorTests.GetAvailableJobAsyncMulithreading [162ms]
  Error Message:
   Assert.Equal() Failure
Expected: 2
Actual:   0
  Stack Trace:
     at Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Tests.Engine.LegacyJobOrchestratorTests.GetAvailableJobAsyncMulithreading() in /home/vsts/work/1/s/components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Engine/LegacyJobOrchestratorTests.cs:line 49
--- End of stack trace from previous location where exception was thrown ---

```

Reply:
Should work now, does it?

Reply:
Yes, it does. Thank you.